java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -370,177 +363,226 @@ public final void visitMemberSelectExpression(MemberSelectExpressionTree tree) {
       if (isSuperOrThisMemberSelect(tree)) {
         scan(tree.identifier());
       } else {
-        super.visitMemberSelectExpression(tree);
-        currentResult = SymbolicBooleanConstraint.UNKNOWN;
+        scan(tree.expression());
+        SymbolicValue value = new SymbolicValue();
+        for (ExecutionState executionState : currentStates) {
+          executionState.pop();
+          executionState.push(value);
+        }
       }
     }
 
     @Override
     public final void visitMethodInvocation(MethodInvocationTree tree) {
-      super.visitMethodInvocation(tree);
-      currentState.invalidateFields();
-      currentResult = SymbolicBooleanConstraint.UNKNOWN;
+      scan(tree.methodSelect());
+      scan(tree.arguments());
+      for (ExecutionState executionState : currentStates) {
+        SymbolicValue[] arguments = new SymbolicValue[tree.arguments().size()];
+        for (int i = arguments.length - 1; i >= 0; i -= 1) {
+          arguments[i] = executionState.pop();
+        }
+        check.onExecutableElementInvocation(executionState, tree, Arrays.asList(arguments));
+        executionState.invalidateFields();
+      }
     }
 
     @Override
-    public final void visitUnaryExpression(UnaryExpressionTree tree) {
-      super.visitUnaryExpression(tree);
-      if (tree.is(Tree.Kind.LOGICAL_COMPLEMENT)) {
-        currentResult = currentResult.negate();
-      } else {
-        if (tree.is(Tree.Kind.POSTFIX_DECREMENT, Tree.Kind.POSTFIX_INCREMENT, Tree.Kind.PREFIX_DECREMENT, Tree.Kind.PREFIX_INCREMENT)) {
-          Symbol.VariableSymbol symbol = extractVariableSymbol(tree.expression());
-          if (symbol != null) {
-            currentState.invalidateRelationsOnValue(new SymbolicValue.SymbolicVariableValue(symbol));
-          }
+    public final void visitNewArray(NewArrayTree tree) {
+      scan(tree.dimensions());
+      scan(tree.initializers());
+      SymbolicValue value = new SymbolicValue();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this instantiation of "SymbolicValue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2440)